node-src/lib/getConfiguration.ts

```diff
@@ -0,0 +1,64 @@
+import { readFile } from 'jsonfile';
+import { ZodError, z } from 'zod';
+import { missingConfigurationFile } from '../ui/messages/errors/missingConfigurationFile';
+import { unparseableConfigurationFile } from '../ui/messages/errors/unparseableConfigurationFile';
+import { invalidConfigurationFile } from '../ui/messages/errors/invalidConfigurationFile';
+
+const configurationSchema = z
+  .object({
+    projectId: z.string(),
+    projectToken: z.string(),
+
+    onlyChanged: z.union([z.string(), z.boolean()]),
+    onlyStoryFiles: z.array(z.string()),
+    onlyStoryNames: z.array(z.string()),
+    untraced: z.array(z.string()),
+    externals: z.array(z.string()),
+    ci: z.boolean(),
+    debug: z.boolean(),
+    junitReport: z.union([z.string(), z.boolean()]),
```

Comment:
Looking at this option, wouldn't the `diagnostics` one be a better fit here for debugging purposes?  Also, wouldn't the`ci` flag impact the builds coming from the addon itself? Unless there's something more to to these that I'm not seeing here.

Reply:
The `junitReport` is used to generate a file consumed by services like circle CI, ala:

<img width="710" alt="image" src="https://github.com/chromaui/chromatic-cli/assets/132554/67fa6f92-865b-4a30-b7f9-dd24d7fa0623">

My aim here is to include all the options that you might want to configure chromatic "permanently" to (eventually) replace the need for CLI flags, outside of one-off or special case invocations of the CLI.

Reply:
@tmeasday I talked with @ghengeveld, and one thing that may have a ripple effect is when a user enables the `ci` flag in the configuration file. For example, the user enables the option in the file and runs a build from the CLI/CI. Then, it links it to the addon and runs another build. As the `ci` option is enabled, it can cause some issues as the second build will not be marked as a local build but as a public build (i.e., from the CLI or CI job) when it's not. I was wondering if we should avoid having it for now or have a detection mechanism for situations like this to prevent issues. Happy to discuss this in more detail.

Reply:
I think you are right, it's not a good flag for the file, let's drop it.